### PR TITLE
Fix issues pointed out by clang-analyzer

### PR DESCRIPTION
**Pull request by mmalecki**
This fixes all clang-analyzer warnings besides `tree.h`-related ones and the one in `tty.c`, which would break Darwin.

We went down from this:
2014-09-27-005350_1920x1080_scrot

To this:
2014-09-27-005347_1920x1080_scrot

### CODE REVIEW

File: src/unix/linux-core.c [hunk 1]
```diff
@@ -743,6 +743,9 @@ int uv_interface_addresses(uv_interface_address_t** addresses,
     (*count)++;
   }
 
+  if (*count == 0)
+    return 0;
```
Comment (saghul) [inline src/unix/linux-core.c:748]: Shouldn't we set *addresses to NULL before returning? Otherwise `uv_free_interface_addresses` would probably crash.

Comment (mmalecki) [inline src/unix/linux-core.c:748]: Good point. We should also indicate that `uv_interface_addresses` will set `addresses` to `NULL` if we have 0 interfaces in our documentation.

Comment (mmalecki) [inline src/unix/linux-core.c:748]: @saghul force pushed that commit

**Comment by mmalecki** [comment]
@saghul I just realized that I opened this PR against `master`, should I reopen against `v1.x`?

**Comment by saghul** [comment]
Please do :-)

**Comment by saghul** [comment]
Closing, if you still want to fix these, please open a PR against the 1,x branch in libuv/libuv. Thanks!

**Comment by mmalecki** [comment]
Ah, damn, completely forgot about this. PR'ing now.

**Pull request closed without merging**